src/pages/debrief/__tests__/debrief.effects.spec.ts

```diff
@@ -53,6 +60,22 @@ describe('Debrief Effects', () => {
       });
     });
 
+    it('should return SET_TEST_STATUS_WRITE_UP & PERSIST_TESTS actions when failed test', (done) => {
+      // Set activity code as failed
+      actions$.next(new testsActions.SetActivityCode('2'));
+      actions$.next(new debriefActions.EndDebrief());
+
+      effects.endDebriefEffect$.subscribe((result) => {
+        if (result instanceof testStatusActions.SetTestStatusWriteUp) {
+          expect(result).toEqual(new testStatusActions.SetTestStatusWriteUp(currentSlotId));
+        }
+        if (result instanceof testsActions.PersistTests) {
+          expect(result).toEqual(new testsActions.PersistTests());
```

Comment:
As discussed, `done()` below these two if cases will not give us coverage of both branches. As we have guarantee of the new action dispatch order in this case, `done()` can be moved into the second if.

Reply:
We could set a counter and track all the assertion and call `done` when we've done all of them?

Reply:
I'll go searching again but IIRC testing for conditional emissions from an effect was unsupported.

Reply:
We haven't updated anything so I imagine the case is much the same now.